src/Rings.jl

```diff
@@ -241,3 +241,49 @@ function is_finite(R::NCRing)
   c == 1 && return true
   throw(NotImplementedError(:is_finite, R))
 end
+
+@doc raw"""
+    is_noetherian(R::Ring)
+
+Check if the ring $R$ is Noetherian.
+```jldoctest
+julia> R, x = polynomial_ring(ZZ, [:x]);
+
+julia> is_noetherian(R)
+true
+```
+"""
+function is_noetherian(R::Ring)
+  throw(NotImplementedError(:is_noetherian, R))
+end
```

Comment:
I wonder: Do we gain anything with this over a regular `MethodError`? I.e. how about we just do this (which indicates a function stub without any methods) ?
```suggestion
function is_noetherian end
```

Reply:
`MethodError`s are cryptic and not helpful for users. They don't know whether they made an error or not.

Reply:
Hmm, are they? Let's see: right now with this PR I get this:
```
julia> R = residue_ring(ZZ, 6)[1]
Residue ring of integers modulo 6

julia> krull_dim(R)
ERROR: function is_noetherian is not implemented for argument
EuclideanRingResidueRing{BigInt}: Residue ring of integers modulo 6

Stacktrace:
 [1] is_noetherian(R::EuclideanRingResidueRing{BigInt})
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:257
 [2] macro expansion
   @ ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Assertions.jl:601 [inlined]
 [3] krull_dim(R::EuclideanRingResidueRing{BigInt})
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:278
 [4] top-level scope
   @ REPL[5]:1
```

If I remove the default method, I get this:
```
julia> R = residue_ring(ZZ, 6)[1]
Residue ring of integers modulo 6

julia> krull_dim(R)
ERROR: MethodError: no method matching is_noetherian(::EuclideanRingResidueRing{BigInt})

Closest candidates are:
  is_noetherian(::Union{AbstractAlgebra.MSeriesRing, SeriesRing})
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:262
  is_noetherian(::Union{LaurentMPolyRing, LaurentPolyRing, MPolyRing, PolyRing})
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:261
  is_noetherian(::AbstractAlgebra.Integers)
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:259
  ...

Stacktrace:
 [1] macro expansion
   @ ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Assertions.jl:601 [inlined]
 [2] krull_dim(R::EuclideanRingResidueRing{BigInt})
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/Rings.jl:276
 [3] top-level scope
   @ REPL[7]:1
```


The main differences:

- the message "ERROR: function is_noetherian is not implemented for argument" is probably indeed a bit more helpful for people new to Julia
  - counterpoint: if you use Julia, you will very quickly start seeing errors of the form `ERROR: MethodError: no method matching ...` elsewhere and need to understand them, and once you do that, is there really a relevant difference?
- the list of "Closest candidates" is maybe helpful for a developer but probably not so much for a user

Reply:
It is mainly for situations where this error will be thrown from within other functions. E.g. `factor` for `R[x]` might throw `factor not implemented for $(typeof(R))`, instead of a `MethodError` with some useless list of closest candidates (the list will be confusing and not helpful for the user).

Reply:
Let's put a pin in this discussion, we can easily change it one way or the other later on.